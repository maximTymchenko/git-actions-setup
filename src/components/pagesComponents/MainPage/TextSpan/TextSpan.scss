@import '../../../../styles/vars';

.text-span {
    display: flex;
    align-items: flex-start;
    justify-content: center;
    position: relative;
    overflow: hidden;
    @include section__size;
    margin: 40px auto 30px;
    padding-bottom: 25px;

    @media screen and (min-width: $mac__13) {
        margin: 30px auto;
    }

    @media screen and (max-width: $tablet) {
        margin-left: 0px;
    }

    @media screen and (max-width: $mobile) {
        margin-top: 0px;
        margin-bottom: 15px;
    }

    &::before {
        position: absolute;
        content: url('../../../../assets/img/waves.svg');
        top: 10px;
        left: -200px;

        @media screen and (max-width: $desktop) {
            left: -320px;
        }

        @media screen and (max-width: $tablet) {
            padding: 90px 0;
            left: -380px;
            top: -120px;
        }

        @media screen and (max-width: $mobile) {
            content: url('../../../../assets/img/medium-waves.svg');
            left: -60px;
            top: -80px;
            transform: rotateY(180deg);
        }
    }

    &::after {
        overflow: hidden;
        position: absolute;
        content: url('../../../../assets/img/waves.svg');
        transform: rotateX(180deg);
        top: 10px;
        right: -200px;

        @media screen and (max-width: $desktop) {
            right: -320px;
        }

        @media screen and (max-width: $tablet) {
            right: -350px;
            bottom: -100px;
            padding: 70px 0;
        }

        @media screen and (max-width: $mobile) {
            content: url('../../../../assets/img/medium-waves.svg');
            right: 0;
            bottom: -60px;
        }
    }

    &__wrapper {
        display: flex;
        flex-direction: column;
        align-items: center;
        max-width: 400px;
        width: 100%;
        margin-bottom: 50px;

        @media screen and (max-width: $mobile) {
            padding: 110px 0;
        }
    }

    &__info {
        max-width: 380px;
        font-family: 'OktaNeue-Regular';
        font-size: 15px;
        text-align: center;
        color: $text__body;
        margin-bottom: 30px;

        @media screen and (max-width: $mobile) {
            width: 320px;
        }
    }
}
