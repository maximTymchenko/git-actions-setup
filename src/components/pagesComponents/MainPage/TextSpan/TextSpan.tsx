import React, { FC } from 'react';
import './TextSpan.scss';
import SectionTitle from '../../../common/SectionTitle';
import Button from '../../../common/Button';
import Calendar from '../../../../assets/img/calendar.svg';

interface ITextSpan {
    title: string;
    children: string;
    type: string;
}

const TextSpan: FC = (props: ITextSpan) => {
    return (
        <section data-testid="text-span" className={`text-span text-span--${props.type}`}>
            <div className="text-span__wrapper">
                <SectionTitle type={'secondary'} className="title" title={props.title} />
                <p className="text-span__info">{props.children}</p>
                <Button type="secondary" desc="Discover all" icon={Calendar}>
                    <a target="_blank" href="https://calendly.com/yuriidrozd">
                        Book a meeting
                    </a>
                </Button>
            </div>
        </section>
    );
};

export default TextSpan;
