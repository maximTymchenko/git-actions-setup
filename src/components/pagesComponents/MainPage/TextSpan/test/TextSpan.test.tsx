import React from 'react';
import TextSpan from '../TextSpan';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(
        <TextSpan
            title="Shall we chat quickly?"
            text="Even a 5-minute intro call could potentially save you countless hours of endless
                    back-and-forth emails."
        />,
    );
    expect(getByTestId('text-span')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer
        .create(
            <TextSpan
                title="Shall we chat quickly?"
                text="Even a 5-minute intro call could potentially save you countless hours of endless
                    back-and-forth emails."
            />,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
