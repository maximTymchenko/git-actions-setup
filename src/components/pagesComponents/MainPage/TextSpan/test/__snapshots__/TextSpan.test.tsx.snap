// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<section
  className="text-span text-span--undefined"
  data-testid="text-span"
>
  <div
    className="text-span__wrapper"
  >
    <h2
      className="title title--secondary"
      data-testid="section-title"
    >
      Shall we chat quickly?
    </h2>
    <p
      className="text-span__info"
    />
    <button
      className="button button--secondary button--undefined button--undefined"
      data-testid="button"
    >
      <a
        href="https://calendly.com/yuriidrozd"
        target="_blank"
      >
        Book a meeting
      </a>
      <img
        alt="Discover all"
        className="button__icon"
        data-testid="button__icon"
        src="test-file-stub"
      />
    </button>
  </div>
</section>
`;
